src/k8s_charm.py

```diff
@@ -111,6 +127,28 @@ def metrics_rules(self) -> Dict[str, Any]:
         """Return a list of metrics rules."""
         return self._scrape.alerts
 
+    @property
+    def dashboards(self) -> list:
+        """Returns an aggregate of all dashboards received by this grafana-agent."""
+        aggregate = {}
+        for rel in self.model.relations["grafana-dashboards-consumer"]:
+            dashboards = json.loads(rel.data[rel.app].get("dashboards"))
+            if "templates" not in dashboards:
+                continue
+            for template in dashboards["templates"]:
+                content = GrafanaDashboard(
+                    dashboards["templates"][template].get("content")
+                )._deserialize()
+                entry = {
+                    "charm": dashboards["templates"][template].get("charm", "charm_name"),
+                    "relation_id": rel.id,
+                    "title": template,
+                    "content": content,
+                }
+                aggregate[template] = entry
+
+        return aggregate.values()
+
```

Comment:
Why aren't we just using the grafana-dashboard library for this? Seems like we are reinventing the wheel here. Also, if we do need to do this, we should put the GrafanaDashboard class in cosl. I don't like using the cos-agent lib for non cos-agent stuff.

Reply:
We need to do this for multiple reasons. The `GrafanaDashboardConsumer` works on a peer relation specific for Grafana and does a whole lot of completely unnecessary shenanigans when all we want to do is take data from N providers and give it to one consumer.